src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerService.cs

```diff
@@ -453,7 +453,24 @@ internal ExpandResponse QueueExpandNodeRequest(ObjectExplorerSession session, st
                                else
                                {
                                    Logger.Verbose($"Expanding {nodePath}");
-                                   nodes = node.Expand(cancelToken, securityToken?.Token, filters).Select(x => x.ToNodeInfo()).ToArray();
+                                   try
+                                   {
+                                       nodes = node.Expand(cancelToken, securityToken?.Token, filters).Select(x => x.ToNodeInfo()).ToArray();
+                                   }
+                                   catch (ConnectionFailureException ex)
+                                   {
+                                       Logger.Error($"Failed to expand node: {ex.InnerException.Message}");

```

Comment:
Are we guaranteed to have an InnerException?

Reply:
No, it's hard to guarantee that we're always going to have an inner exception since the exception is being thrown from external code. That external exception is caught in the snippet linked below and a second one is thrown when the `context?.Database` is accessed.

https://github.com/microsoft/sqltoolsservice/blob/a12244f3d0c1377df8ee30fc36a12781ac547677/src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/DatabaseTreeNode.cs#LL87C1-L87C1